src/types/redis_bloom_chain.h

```diff
@@ -57,18 +64,21 @@ class BloomChain : public Database {
  public:
   BloomChain(engine::Storage *storage, const std::string &ns) : Database(storage, ns) {}
   rocksdb::Status Reserve(const Slice &user_key, uint32_t capacity, double error_rate, uint16_t expansion);
-  rocksdb::Status Add(const Slice &user_key, const Slice &item, BloomFilterAddResult *ret);
-  rocksdb::Status MAdd(const Slice &user_key, const std::vector<Slice> &items, std::vector<BloomFilterAddResult> *rets);
-  rocksdb::Status Exists(const Slice &user_key, const Slice &item, bool *exist);
-  rocksdb::Status MExists(const Slice &user_key, const std::vector<Slice> &items, std::vector<bool> *exists);
+  rocksdb::Status Add(const Slice &user_key, const std::string &item, BloomFilterAddResult *ret);
+  rocksdb::Status MAdd(const Slice &user_key, const std::vector<std::string> &items,
+                       std::vector<BloomFilterAddResult> *rets);
+  rocksdb::Status InsertCommon(const Slice &user_key, const std::vector<std::string> &items,
```

Comment:
Why is it called `InsertCommon` instead of `Insert`?

Reply:
It's named as RedisBloom do. I think it indicate not only `Insert` but `Madd` share this function.

Reply:
I guess this is extended from RedisBloom, lol

We can separate underlying impl and interface @zncleon

Reply:
But the main logic of these two function is same. If we separate them, we would have many redundant codes. @mapleFU

Reply:
Got it.